props/saber_BC_buttons.h

```diff
@@ -386,31 +1161,254 @@ EFFECT(volmax);     // for maximum volume reached
+  }
+
+  void next() override {
+    highlighted_blade_.Stop(current_blade());
+    SPEC::MenuBase::next();
+    highlighted_blade_.Start(current_blade());
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade() << "\n";
+  }
+
+  void prev() override {
+    highlighted_blade_.Stop(current_blade());
+    SPEC::MenuBase::prev();
+    highlighted_blade_.Start(current_blade());
+    PVLOG_NORMAL << "** Highlighting blade: " << current_blade() << "\n";
+  }
+
+  void say() override {
+    mode::getSL<SPEC>()->SayBlade();
+    mode::getSL<SPEC>()->SayWhole(current_blade());
+  }
+
+  void select() override {
+    // Set the current blade to send and push to ChangeBladeLengthMode
+    mode::menu_current_blade = current_blade();
+    highlighted_blade_.Stop(current_blade());
+    pushMode<typename SPEC::ChangeBladeLengthMode>();
+  }
+};
+
+template<class SPEC>
+// struct BCChangeBladeLengthBlade1 : public SPEC::ChangeBladeLengthMode {
+struct BCChangeBladeLengthBlade1 : public mode::ChangeBladeLengthBlade1<SPEC> {
+  // virtual int blade() { return mode::menu_current_blade; }
+
+  int steps_per_revolution() override {
+    return 30;  // adjust for sensitivity
+  }
+  void select() override {
+    prop_SetBladeLength(this->blade(), this->getLength());
+    prop_SaveState();
+    mode::getSL<SPEC>()->SaySave();
+    popMode();
+  }
+  void update() override {
+    hybrid_font.PlayPolyphonic(&SFX_volup);
+    this->say_time_ = Cyclint<uint32_t>(millis()) + (uint32_t)(SaberBase::sound_length * 1000) + 300;
+    if (!this->say_time_) this->say_time_ += 1;
+    this->fadeout(SaberBase::sound_length);
+  }
+};
+#endif  // DYNAMIC_BLADE_LENGTH
+
+template<class SPEC>
+struct BCMenuSpec {
+  typedef BCVolumeMode<SPEC> BCVolumeMenu;
+#ifdef DYNAMIC_BLADE_LENGTH
+  typedef BCSelectBladeMode<SPEC> BCSelectBladeMenu;
+  typedef mode::ShowLengthStyle<SPEC> ShowLengthStyle;
+  typedef BCChangeBladeLengthBlade1<SPEC> ChangeBladeLengthMode;
+#endif  // DYNAMIC_BLADE_LENGTH
+  typedef mode::SelectCancelMode SelectCancelMode;
+  typedef mode::SteppedMode<SPEC> SteppedMode;
+  typedef mode::SteppedModeBase<SPEC> SteppedModeBase;
+  typedef mode::MenuBase<SPEC> MenuBase;
+  typedef SoundLibraryV2 SoundLibrary;
+};
+
+#ifndef MENU_SPEC_TEMPLATE
+  void Setup() {
+    MKSPEC<BCMenuSpec>::SoundLibrary::init();
+  }
+#endif
+
+class DelayTimer {
+public:
+  DelayTimer() : triggered_(false), trigger_time_(0), duration_(0) {}
+
+  void trigger(uint32_t duration) {
+      triggered_ = true;
+      trigger_time_ = millis();
+      duration_ = duration;
+  }
+  void stop() {
+      PVLOG_DEBUG << "** STOPPING timer.\n";
+      triggered_ = false;
+  }
+  bool timerCheck() {
+      if (!triggered_) return false;
+      if (millis() - trigger_time_ > duration_) {
+          triggered_ = false;  // Reset the timer flag
+          return true;  // Timer has elapsed
+      }
+      return false;  // Timer is still running
+  }
+
+  bool isTimerRunning() const {
```

Comment:
I think this function should call timerCheck() (and not be const)

Reply:
That is more accurate. How about losing the triggered_ variable too.
See update.

Reply:
Actually that doesn't work, I'm getting premature resets.
  This works instead though.
  ```cpp
  bool isTimerRunning() {
    return trigger_time_ != 0 && (millis() - trigger_time_ < duration_);
  }
  ```

Reply:
What I was imagining was:
bool isTimerRunning() {
  timerCheck();
 return triggered_;
}

Reply:
ohhh. ok yeah that works. 
Updated.